### PR TITLE
refactor conversations needed into separate function

### PR DESCRIPTION
**Pull request by mus1cholic**
**Patch description**

Moved conversations_needed logic code to separate function to be easily refactored

**Testing steps**

See CircleCI tests

**Other information**

### CODE REVIEW

File: parlai/crowdsourcing/tasks/model_chat/model_chat_blueprint.py [hunk 1]
```diff
@@ -453,6 +447,17 @@ def __init__(
             }
         )
 
+    def _process_conversations_needed(self, args: "DictConfig") -> Dict[str, str]:
```
[the remark this answers is not in the document]
Comment (mus1cholic) [reply parlai/crowdsourcing/tasks/model_chat/model_chat_blueprint.py]: done

File: parlai/crowdsourcing/tasks/model_chat/model_chat_blueprint.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from dataclasses import dataclass, field
 from threading import Semaphore, Condition
 from typing import Any, Dict, List, Optional, TYPE_CHECKING
+from typing_extensions import IntVar
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/model_chat/model_chat_blueprint.py:15]: You can take this out :)

File: parlai/crowdsourcing/tasks/model_chat/model_chat_blueprint.py [hunk 3]
```diff
@@ -453,6 +448,17 @@ def __init__(
             }
         )
 
+    def _process_conversations_needed(self, args: "DictConfig") -> Dict[str, int]:
+        # Set the number of conversations needed
```
Comment (EricMichaelSmith) [inline parlai/crowdsourcing/tasks/model_chat/model_chat_blueprint.py:452]: Nitpick: can you turn this into a docstring? i.e.:
```
"""
Set the number of conversations needed
"""
```

**Review by EricMichaelSmith — COMMENTED** [review]
Thanks for the update! Some last nitpicky things

**Review by EricMichaelSmith — APPROVED** [review]
Yeah looks great! Thanks for the changes, approved :)

**Pull request merged**